test/io/test_io.py

```diff
@@ -545,26 +544,6 @@ def test_iat_claim(defaultenv):
             time.sleep(0.1)
 
 
-def test_app_settings(defaultenv):
-    """
-    App settings should not reset when the db pool times out.
```

Comment:
This test still passes, but it no longer makes sense.

Reply:
Right. So really the whole app.settings feature was meant to add GUCs at the session level(https://github.com/PostgREST/postgrest/pull/1063) but those got lost on timeouts. So I made a lazy fix(https://github.com/PostgREST/postgrest/pull/1169, [this test was in bash at that time](https://github.com/PostgREST/postgrest/pull/1169/files#diff-20d0d2179c4a0dea3feea7ebcc9627b8296ef717828184176520808957f36906R146-R165)) to move them to the transaction level instead - which is wasteful considering they don't change on each transaction.

Now that we don't have timeouts, perhaps we can move `app.settings` to the session level. This would also pair up with the `statement_timeout` feature [discussed below](https://github.com/PostgREST/postgrest/pull/2391#discussion_r931279472), we could reuse the same code to set it at the session level ourselves.

Reply:
I don't quite follow the session vs transaction level setting, if you think that's a change we should make would you file an issue so we don't lose it? I think for this PR we don't need to make a change (and I've updated the test now so that it's in some way still useful).